mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,8 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+        require(indices.size == 0 || indices(0) >= 0,
```

Comment:
I think the sparse vector constructor doesn't verify its input since it's created so frequently. But this seems like a reasonable place to check input, when loading from libsvm. I was mistaken; this will silently fail if the libsvm format is wrong and the indices aren't sorted. So I suppose your original check is fine but if the ordering of the indices is also checked.